test/unit/helpers/multi.spec.ts

```diff
@@ -0,0 +1,31 @@
+import * as path from "path";
+import * as util from "../../util";
+import * as tstl from "../../../src";
+import { unsupportedMultiHelperFunctionPosition } from "../../../src/transformation/utils/diagnostics";
+
+const multiProjectOptions: tstl.CompilerOptions = {
+    types: [path.resolve(__dirname, "../../../helpers")],
+};
+
+test.each<[string, any]>([
+    ["return $multi();", undefined],
+    ["return $multi(1);", 1],
+    ["const ar = [1]; return $multi(...ar);", 1],
```

Comment:
What should it do if there are multiple values being spread, i.e. `const ar = [1, 2, 3]; return $multi(...ar, ...ar);`?

Also would be nice to test spread over function returning multiple values (`$multi(...unpack([1, 2, 3]), ...unpack([4, 5, 6]))`)

Reply:
Currently it works like this which I think is pretty reasonable

Source:
```ts
function x(...args: any[]) {
  return $multi(...args, ...args);
}

function y() {
  return $multi(...x("a", "b", "c"));
}

console.log(x(1, 2, 3));
console.log(y());
```

Lua:

```lua
--[[ Generated with https://github.com/TypeScriptToLua/TypeScriptToLua ]]
require("lualib_bundle");
function x(self, ...)
    local args = {...}
    return __TS__Unpack(
        __TS__ArrayConcat(
            { __TS__Unpack(args) },
            { __TS__Unpack(args) }
        )
    )
end
function y(self)
    return x(nil, "a", "b", "c")
end
print(x(nil, 1, 2, 3))
print(y(nil))
```

Output:
```
1       2       3       1       2       3
a       b       c       a       b       c
```

Reply:
What do you think about limiting it to just regular values and spread of `MultiReturn` as the last element? I see `$multi` as a helper that allows to express Lua syntax that isn't expressible in TS, so doing more stuff might be quite a surprising behavior

Reply:
Interesting, I don't see much of a problem limiting the spread expressions used in `$multi`, could you provide some code examples on how this would work?

Reply:
```ts
return $multi(1, 2, 3);                                 // return 1, 2, 3
return $multi(1, 2, 3, ...string.byte("abc", 1, 3));    // return 1, 2, 3, string.byte("abc", 1, 3)
return $multi(1, 2, 3, ...unpack([1, 2, 3]));           // return 1, 2, 3, unpack({1, 2, 3})
return $multi(1, 2, 3, ...[1, 2, 3]);                   // error
return $multi(1, ...string.byte("abc", 1, 3), 2, 3);    // error
((...args: any[]) => $multi(1, 2, 3, ...args));         // error
((...args: Vararg<any[]>) => $multi(1, 2, 3, ...args)); // return 1, 2, 3, ...
```